tools/serverpod_cli/bin/serverpod_cli.dart

```diff
@@ -43,13 +44,17 @@ void main(List<String> args) async {
     () async {
       try {
         await _main(args);
+      } on ExitException catch (e) {
+        exit(e.exitCode);
       } catch (error, stackTrace) {
         // Last resort error handling.
```

Comment:
We may want to wait for the analytics to finish here too (i.e., refer to how we normally end the program).